ably/realtime_conn_spec_test.go

```diff
@@ -467,6 +467,7 @@ func TestRealtimeConn_RTN15c2(t *testing.T) {
 		t.Fatal("expected message after connection recovery; got none")
 	}
 
+	<-stateChanges
```

Comment:
If we wait on a channel without a timeout, the test will just hang forever if we ever have a regression.

(You might want to backport [these helpers](https://github.com/ably/ably-go/commit/6abf09a4a86d237e7c006fe761516b6c0ce67a02) I added to the 1.2 branch. So you could do e. g. `ablytest.Soon.Recv(t, nil, stateChanges, t.Fatal)` instead of a big, noisy `select`.)

Reply:
I think since we will be on `1.2` branch after this PR I will address this there in a separate PR to cleanup the tests.